app/src/main/java/org/dhis2/usescases/enrollment/EnrollmentPresenterImpl.kt

```diff
@@ -438,22 +439,6 @@ class EnrollmentPresenterImpl(
             }
 
             if (field !is SectionViewModel && field !is DisplayViewModel) {
-                val isUnique =
-                    d2.trackedEntityModule().trackedEntityAttributes()
-                        .uid(field.uid()).blockingGet()?.unique() ?: false
-                var uniqueValueAlreadyExist: Boolean
-                if (isUnique && field.value() != null) {
-                    uniqueValueAlreadyExist =
-                        d2.trackedEntityModule()
-                        .trackedEntityAttributeValues()
-                        .byTrackedEntityAttribute()
-                        .eq(field.uid())
-                        .byValue().eq(field.value())
-                        .blockingGet().size > 1
-                    if (uniqueValueAlreadyExist) {
-                        uniqueFields[field.uid()] = field.label()
-                    }
-                }
```

Comment:
Hello @Balcan , this would work if you don't change the focus and click on save button?

Reply:
Yes, I changed the update of the unique field list to when we show the dialog when attempting to save a value